ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -126,6 +135,17 @@ const Footer = () => {
     return false;
   });
 
+  const isSIWE = isSIWESignatureRequest(currentConfirmation);
+  const isPermit = isPermitSignatureRequest(currentConfirmation);
+  const isPermitSimulationShown = isPermit && useTransactionSimulations;
+
+  const isConfirmDisabled =
+    (!isScrollToBottomCompleted && !isSIWE && !isPermitSimulationShown) ||
+    ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
+    mmiSubmitDisabled ||
+    ///: END:ONLY_INCLUDE_IF
+    hardwareWalletRequiresConnection;
```

Comment:
It will be nice to create context for scroll functionality and move these logic to it - so that the logic does not goes to components like footer.

Reply:
Great suggestion!

I created a new, enhancement issue to address this here: https://github.com/MetaMask/metamask-extension/issues/26970